### PR TITLE
3.8: Use lambdas instead of boost::bind

### LINKED ISSUE #3720: Investigate: `_1` placeholder unknown on very modern platforms
Occured on a F33 build:

https://download.copr.fedorainfracloud.org/results/marcusmueller/gnuradio/fedora-33-x86_64/01623243-gnuradio/builder-live.log.gz

### PR DESCRIPTION
**Pull request by marcusmueller**
This is mostly a backport of master functionality.

Fixes #3720

Discussion there.

### CODE REVIEW

File: gr-uhd/lib/usrp_block_impl.cc [hunk 1]
```diff
@@ -137,12 +137,13 @@ usrp_block_impl::usrp_block_impl(const ::uhd::device_addr_t& device_addr,
...
+                    [this](pmt::pmt_t msg) { this->msg_handler_command(msg); });
 
-// cuz we lazy:
+    // because we're highly efficient and adverse to work
+    namespace ph = std::placeholders;
 #define REGISTER_CMD_HANDLER(key, _handler) \
-    register_msg_cmd_handler(key,           \
-                             boost::bind(&usrp_block_impl::_handler, this, _1, _2, _3))
+    register_msg_cmd_handler(               \
+        key, std::bind(&usrp_block_impl::_handler, this, ph::_1, ph::_2, ph::_3))
```
Comment (jdemel) [inline gr-uhd/lib/usrp_block_impl.cc:146]: Are you able to use lambdas here too? I'd argue that would make it more consistent.

**Review by jdemel — COMMENTED** [review]
Most of it looks good. Just one comment.

Are there any `boost::bind` occurrences left in the code base?

File: gr-uhd/lib/usrp_source_impl.cc [hunk 2]
```diff
@@ -57,8 +57,10 @@ usrp_source_impl::usrp_source_impl(const ::uhd::device_addr_t& device_addr,
 
     _samp_rate = this->get_samp_rate();
     _samps_per_packet = 1;
-    register_msg_cmd_handler(cmd_tag_key(),
-                             [this](const pmt::pmt_t& val, int chan, const pmt::pmt_t& msg) { this->_cmd_handler_tag(val); });
+    register_msg_cmd_handler(
+        cmd_tag_key(), [this](const pmt::pmt_t& val, int chan, const pmt::pmt_t& msg) {
+            this->_cmd_handler_tag(val);
+        });
```
Comment (jdemel) [inline gr-uhd/lib/usrp_source_impl.cc:63]: I assume with this change you can remove the `functional` include at the top of the file.

**Comment by mbr0wn** [comment]
@marcusmueller can you rebase? You could also split up the UHD and non-UHD parts.

**Comment by mbr0wn** [comment]
@marcusmueller I ported the first three commits to maint-3.8, but can you please rebase and resolve the rest of the issues.

**Comment by mbr0wn** [comment]
Superseded by https://github.com/gnuradio/gnuradio/pull/4103

**Pull request closed without merging**